src/bci_build/package/python/README.md.j2

```diff
@@ -0,0 +1,57 @@
+# {{ image.pretty_name }} Container Image
+
+{% include 'badges.j2' %}
+
+## Description
+
+[Python](https://www.python.org/) is an interpreted, interactive, object-oriented, open-source programming language. It incorporates modules, exceptions, dynamic typing, high-level dynamic data types, and classes. It provides interfaces to many system calls, libraries, and various window systems, and it is extensible in C or C++. It is also usable as an extension language for applications that require programmable interfaces.
+
+## Usage
+
+To deploy an application, install dependencies, copy the sources, and configure the application's main script:
+
+```Dockerfile
+FROM {{ image.pretty_reference }}
+
+WORKDIR /app
+
+COPY requirements.txt ./
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
+
+CMD [ "python", "./main-script.py" ]
+```
+
+Build and run the container image:
+
+```ShellSession
+$ podman build -t my-python-app .
+$ podman run -it --rm my-python-app
+```
+
+To run a single script inside a container, use the following command:
+
+```ShellSession
+$ podman run --rm -v "$PWD":/app:Z -w /app {{ image.pretty_reference }} python3 script.py
+```
+
+## Additional tools
+
+The following tools are included in the image:
+
+{% set has_pipx = image.is_opensuse or (image.os_version.is_sle15 and image.version == '3.11') -%}
```

Comment:
TBH I don't prefer adding a set "has_pipx" if the condition is only used in one place later), but I understand this is a personal preference topic. 

However, I wonder if we can move the "has_pipx" into an attribute on the "image" somehow and set it from python.py so that we have the logic only in one place and not duplicated in the README? we're going to forget to update it in two places for sure.

Reply:
This is a long check, it looks better if we use `%set` IMHO.

To be honest, I had the same question, there's a piece in the code that does the same check, and now the same piece in the README, which can be forgotten one day.

The container is `DevelopmentContainer` and it does not make sense to add a Python-exclusive variable here, we could create a `PythonDevelopmentContainer` and set it there.

I could replace this part with the attribute `package_list`, but this might bring packages we don't want, but at least you change the package list in a single place (python code), which is possibly a better idea. We could create a `pretty_package_list` to be used in the README and perhaps clean any unwanted packages if needed.

@dcermak @dirkmueller any better ideas?

Reply:
pretty_package_list sounds like a good idea (splitting package_list into those that would be listed in the readme by specifying them as pretty_package_list and any extra stuff as package_list)

alternatively simply store it as an extra kwargs attribute that we can query. that also works for this one case. I don't think we should invent too many specialized classes but that works also to store an attribute. it's pretty much the same idea.

Reply:
I'd first try if you can perform a `setattr(CONTAINER_INSTANCE, has_pipx, True/False)` or an equivalent of that (it might be that the `dataclass` will block you from doing that). The attribute will then be available in the template.

I'm not a huge fan of either another subclass nor of `pretty_package_list` as I don't see too many use cases for that atm.

Reply:
It is not possible to set attributes because the code where all the logic lives in `_get_python_kwargs`.
I can't add unknown `kwargs` in this function, and I don't have access to the `DevelopmentContainer` instance here. To add it to `DevelopmentContainer` I would need to duplicate code for each container instance, or create yet another function such as `_get_has_pipx`, which looks uglier than other options IMHO.

`pretty_package_list` does not work also, because `package_list` includes the `python311` package as well, which I have no way of removing because there's no variable to match.

No solution seems to work good enough.

For `pretty_package_list` I can remove any packages from `lifecycle_data_pkg` or any packages that are not `PackageType.IMAGE`. To remove the main packages such as `python311` or `python311-devel` I would need to have these in an attribute somewhere else that is not `package_list`.

An option is to create an attribute named `additional_package_list`, which is used by the README, but could also be used by the generator. For example, `package_list` can take the required packages, while `additional_package_list` takes the packages that add extra functionality, such as: make, git, pipx, or else.

The golang and tomcat images also have extra packages in the README. The PHP image has also extra packages that are not mandatory, such as composer. In theory, we could make use of `pretty_package_list` or `additional_package_list` in other images.

Reply:
@dirkmueller @dcermak WDYT? All options have drawbacks.

Reply:
If we want to start listing packages in the READMEs, then `pretty_package_list` is probably the best approach. Do you want to create a PR for that @alexandrevicenzi?

Reply:
@dcermak `pretty_package_list` does not work unless we add an `additional_package_list` or change the `Package` class, this seems a massive change to how packages are defined. Or we list all packages, but the rule so far has been to list extra packages only, which are often useful to add a bit more functionality to a container. The user does not need to know all packages in the image, but it is useful to know that make, git, or pipx is available.

I'm more inclined to `PythonDevelopmentContainer`, which is not disruptive to other containers, and given that we have a `python.py` file where we define the Python container image it would not be that weird to extend `DevelopmentContainer`.

Reply:
I think the suggestion is to go with "pretty_package_list" and "additional_package_list" . this way the readme's can tag them accordingly. we just need to change a few places to include both package lists.

Reply:
I have added #1267 and #1268 as a potential solution, we should pick one. Both proposals should have little side effects, one proposal is a lot simpler than the other.

Reply:
Since #1267 was merged, this has been fixed and rebased.